modules/auxiliary/server/capture/ldap.rb

```diff
@@ -59,328 +53,26 @@ def initialize(info = {})
-                result_code,
-                '',
-                Net::LDAP::ResultStrings[result_code],
-                Net::LDAP::PDU::BindResult
-              )
-            when Net::LDAP::PDU::SearchRequest
-              filter = Net::LDAP::Filter.parse_ldap_filter(pdu.search_parameters[:filter])
-              attrs = pdu.search_parameters[:attributes].empty? ? :all : pdu.search_parameters[:attributes]
-              res = search_res(filter, pdu.message_id, attrs)
-              if res.nil? || res.empty?
-                result_code = Net::LDAP::ResultCodeNoSuchObject
-              else
-                client.write(res)
-                result_code = Net::LDAP::ResultCodeSuccess
-              end
-              service.encode_ldap_response(
-                pdu.message_id,
-                result_code,
-                '',
-                Net::LDAP::ResultStrings[result_code],
-                Net::LDAP::PDU::SearchResult
-              )
-            when Net::LDAP::PDU::UnbindRequest
-              client.close
-              nil
-            else
-              if suitable_response(pdu.app_tag)
-                result_code = Net::LDAP::ResultCodeUnwillingToPerform
-                service.encode_ldap_response(
-                  pdu.message_id,
-                  result_code,
-                  '',
-                  Net::LDAP::ResultStrings[result_code],
-                  suitable_response(pdu.app_tag)
-                )
-              else
-                client.close
-              end
-            end
-
-      on_send_response(client, res) unless res.nil?
-    rescue StandardError => e
-      client.close
-      print_error("Failed to handle LDAP request due to #{e}")
-    end
-  end
-
-  def process_ntlm_hash(arg = {})
-    ntlm_ver = arg[:ntlm_ver]
-    lm_hash = arg[:lm_hash]
-    nt_hash = arg[:nt_hash]
-    unless ntlm_ver == NTLM_CONST::NTLM_V1_RESPONSE || ntlm_ver == NTLM_CONST::NTLM_2_SESSION_RESPONSE
-      lm_cli_challenge = arg[:lm_cli_challenge]
-      nt_cli_challenge = arg[:nt_cli_challenge]
-    end
-    domain = Rex::Text.to_ascii(arg[:domain])
-    user = Rex::Text.to_ascii(arg[:user])
-    host = arg[:name]
-
-    captured_time = Time.now.to_s
-    case ntlm_ver
-    when NTLM_CONST::NTLM_V1_RESPONSE
-      if NTLM_CRYPT.is_hash_from_empty_pwd?({
-        hash: [nt_hash].pack('H*'),
-        srv_challenge: @challenge,
-        ntlm_ver: NTLM_CONST::NTLM_V1_RESPONSE,
-        type: 'ntlm'
-      })
-        print_status('NLMv1 Hash correspond to an empty password, ignoring ... ')
-        return
-      end
-      if lm_hash == nt_hash || lm_hash == '' || lm_hash =~ /^0*$/
-        lm_hash_message = 'Disabled'
-      elsif NTLM_CRYPT.is_hash_from_empty_pwd?({
-        hash: [lm_hash].pack('H*'),
-        srv_challenge: @challenge,
-        ntlm_ver: NTLM_CONST::NTLM_V1_RESPONSE,
-        type: 'lm'
-      })
-        lm_hash_message = 'Disabled (from empty password)'
+  def on_processed_pdu(processed_data)
```

Comment:
This is overriding default behavior instead of registering to receive processed_data for certain types of requests.  This design places requirements on module writers to provide a default `no-op` case that does not need to exists at this layer.